### PR TITLE
SDL_OpenURL (macOS): try to open path if the url cannot be opened

### PR DESCRIPTION
**Pull request by superzazu**
Hello,

This MR updates macOS implementation of SDL_OpenURL so that we can also open paths like "/Users/Bob/Documents".

## Description

Current implementation of macOS SDL_OpenURL can only open URLs such as `https://www.libsdl.org`, `scheme://something` or `file:///Users/Bob/Documents`.

However, some libraries for Mac (for example, PHYSFS) return Unix paths like `/Users/Bob/Documents`. A simple concatenation `"file://" + path` cannot work be cause we can end up with an invalid URL, for example when the path has a space in it (we would end up with `file:///Users/Bob/Library/Application Support/MyApp`).

In this MR, if the url cannot be opened, we try to open the path using another method: creating the NSURL with "fileURLWithPath" (which is [available since macOS 10.0](https://developer.apple.com/documentation/foundation/nsurl/1410828-fileurlwithpath)).

### CODE REVIEW

**Comment by slime73** [comment]
This looks good to me now. It's also worth noting that `[NSURL URLWithDataRepresentation:]` plus `[NSData dataWithBytes]` should work too (as inputs to `[NSWorkspace openURL]`), but either option is good I think.

While I suspect iOS will need the same treatment we'd need to spend time making a test case, which seems slightly more annoying than on macOS because of the limited filesystem etc. So I don't think it needs to be done for this PR.

File: src/misc/macosx/SDL_sysurl.m [hunk 1]
```diff
@@ -27,10 +27,10 @@
 SDL_SYS_OpenURL(const char *url)
 { @autoreleasepool
 {
-    NSString *nsstr = [NSString stringWithUTF8String:url];
-    NSURL *nsurl = [NSURL URLWithString:nsstr];
-    return [[NSWorkspace sharedWorkspace] openURL:nsurl] ? 0 : -1;
+	CFURLRef cfurl = CFURLCreateWithBytes(NULL, (UInt8 *) url, SDL_strlen(url), kCFStringEncodingUTF8,  NULL);
```
Comment (slime73) [inline src/misc/macosx/SDL_sysurl.m:30]: This should probably be `const UInt8 *` rather than the mutable `UInt8 *`.

Comment (slime73) [reply src/misc/macosx/SDL_sysurl.m:30]: Indentation is 4 spaces in SDL's codebase.

**Comment by slouken** [comment]
Can we add an iOS implementation in this patch as well?

Thanks!

File: src/misc/macosx/SDL_sysurl.m (hunk 1 above)
Comment (superzazu) [reply src/misc/macosx/SDL_sysurl.m]: I pushed a commit that fixes your two comments. I still have to add the iOS implementation (LSOpenCFURLRef is not available in iOS)

**Comment by icculus** [comment]
Going to put this in the 2.0.18 milestone for now.

Can we meaningfully open a `file:///` URL on iOS? The files you can access in your app aren't going to be accessible to Safari, etc, right?

**Comment by icculus** [comment]
> Can we meaningfully open a file:/// URL on iOS? The files you can access in your app aren't going to be accessible to Safari, etc, right?

I'm not a careful reader this week; the idea is to send it as a data URL. I'm inclined to say this isn't worth it, though.

**Pull request merged**